fiksi/src/analyze/graph/recursive_assembly.rs

```diff
@@ -442,103 +442,162 @@ pub(crate) fn decompose<const D: i16>(
     recombination_plan
 }
 
-/// Find an arbitrary, minimal dense subgraph.
+/// Find an arbitrary, minimum dense subgraph.
 ///
-/// This searches for a minimal dense subgraph such that
+/// This searches for a minimum dense subgraph such that
 /// `(elements' degrees of freedom) - (constraints' valencies) < D + 1`. See also "Finding Solvable
 /// Subsets of Constraints Graph" (1997) by C. M. Hoffmann, A. Lomonosov, and M. Sitharam.
 ///
-/// The subgraph is minimal in the sense that it contains no proper subgraph upholding that
-/// condition. Trivial subgraphs of single elements are not considered.
+/// The subgraph is minimum in the sense that it is the smallest subgraph containing no proper
+/// subgraph upholding that condition. Trivial subgraphs of single elements are not considered.
 ///
-/// TODO: this currently does an exhaustive search, actually finding optimal *minimum* dense
-/// subgraphs, which is very slow. This should be replaced with a smarter algorithm, as it's not
-/// necessary to find optimal solutions.
-fn dense<const D: i16>(
+/// This does an exhaustive search, which is very slow for every moderately-sized graphs. This
```

Comment:
"every" here also reads oddly.

Reply:
Typo! should be "even," and probably better as "which is very slow even for moderate-sized graphs."